streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -1408,20 +1409,29 @@ public void shouldThrowStreamsExceptionUsingDefaultExceptionHandler() {
     public void shouldDropRecordExceptionUsingAlwaysContinueExceptionHandler() {
         try (final ErrorStringSerializer errorSerializer = new ErrorStringSerializer()) {
             final RecordCollector collector = newRecordCollector(new AlwaysContinueProductionExceptionHandler());
+            final MetricName metricName = new MetricName(
+                    "dropped-records-total",
+                    "stream-task-metrics",
+                    "The total number of dropped records",
+                    mkMap(
+                            mkEntry("thread-id", Thread.currentThread().getName()),
+                            mkEntry("task-id", taskId.toString())
+                    ));
+
+            final Map<MetricName, Metric> mockMetricsMap = new HashMap<>();
+            final Metric metric = Mockito.mock(Metric.class);
+
+            when(metric.metricValue()).thenReturn(1.0);
+            mockMetricsMap.put(metricName, metric);
+            doReturn(mockMetricsMap).when(mockStreamsMetrics).metrics();
+
```

Comment:
Here the code also just verifies the mock setup, it does not verify any production code in `RecordCollectorImpl`.